### ISSUE TITLE
[Bug] Missing downloads/chromedriver.xml

### ISSUE THREAD

**Issue by Chris-Johnston** [cnishina/git-web-launcher#1]
This looks like an awesome project, something I've wanted/needed for a while.

I'm using Linux Mint 18. All the commands executed were in the root directory of a valid Git repo (w/ remote origin pointing to GitHub).

```bash
$ web
curl -o /usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver.xml https://chromedriver.storage.googleapis.com/
(node:11982) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver.xml'
    at Object.openSync (fs.js:443:3)
    at Object.writeFileSync (fs.js:1163:35)
    at Object.<anonymous> (/usr/local/lib/node_modules/github-web-launcher/node_modules/webdriver-manager-replacement/dist/lib/provider/utils/cloud_storage_xml.js:32:16)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/github-web-launcher/node_modules/webdriver-manager-replacement/dist/lib/provider/utils/cloud_storage_xml.js:4:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:11982) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:11982) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The `downloads` directory was missing, so I made a directory and `chown`'d it with the same values as the other directories under `/usr/local/lib/node_modules/github-web-launcher/`

Afterwards, launching as non-root results in this error:

```bash
$ web
curl -o /usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver.xml https://chromedriver.storage.googleapis.com/
(node:16107) UnhandledPromiseRejectionWarning: Error: EACCES: permission denied, open '/usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver.xml'
    at Object.openSync (fs.js:443:3)
    at Object.writeFileSync (fs.js:1163:35)
    at Object.<anonymous> (/usr/local/lib/node_modules/github-web-launcher/node_modules/webdriver-manager-replacement/dist/lib/provider/utils/cloud_storage_xml.js:32:16)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/github-web-launcher/node_modules/webdriver-manager-replacement/dist/lib/provider/utils/cloud_storage_xml.js:4:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:16107) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:16107) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Running `sudo web` results in:

```bash
$ sudo web
curl -o /usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver_linux64.zip https://chromedriver.storage.googleapis.com/2.40/chromedriver_linux64.zip
(node:16505) UnhandledPromiseRejectionWarning: Error: ELOOP: too many symbolic links encountered, chmod '/usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver_2.40'
    at Object.chmodSync (fs.js:996:3)
    at Object.changeFilePermissions (/usr/local/lib/node_modules/github-web-launcher/node_modules/webdriver-manager-replacement/dist/lib/provider/utils/file_utils.js:175:12)
    at ChromeDriver.<anonymous> (/usr/local/lib/node_modules/github-web-launcher/node_modules/webdriver-manager-replacement/dist/lib/provider/chromedriver.js:59:26)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/github-web-launcher/node_modules/webdriver-manager-replacement/dist/lib/provider/chromedriver.js:4:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:16505) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:16505) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

If I delete this `downloads` directory, and try running `sudo web` again:

```bash
$ sudo web
curl -o /usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver.xml https://chromedriver.storage.googleapis.com/
curl -o /usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver_linux64.zip https://chromedriver.storage.googleapis.com/2.40/chromedriver_linux64.zip
/usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver_2.40
/usr/local/lib/node_modules/github-web-launcher/downloads/chromedriver
```

This hangs indefinitely, and Chrome nor my default browser do not appear.

Hope this helps. Please let me know if you need any extra information.

**Comment by cnishina** [comment]
Thanks for the first issue! I was thinking "this is the project no one realized they wanted" and "why doesn't this exist"? 

Anyway, I threw this together as a proof of concept and did a release yesterday to show it off to a few coworkers. It is all still very new and broken. 😛😛😛I literally said "works on my machine" and did an "npm publish"

I hope to nail down some stability by a 1.0.0 release. 😄 Along with tests.

Some issues with `webdriver-manager-replacement`. I'm rewriting my `webdriver-manager` repo and have been playing around with symbolic linking. I have encountered it in some of my integration tests. I might go back to writing everything to file because I'm not sure how it will work in Windows.

I think more verbose logging will help to figure out what's going on here (when global node requires sudo permissions). It also looks like I am missing some catches on my native promises.

Again, thanks for the issue. _Working on it!_

**Comment by cnishina** [comment]
@Chris-Johnston I wonder what version of node you are using. If you are using node maintained by the operating system package manager and need to use sudo, I am not sure if you can start Chrome as root. https://unix.stackexchange.com/questions/175967/how-to-run-google-chrome-as-root-in-linux

Alternatively, you can use nvm to download versions of node and globally install them as your user without sudo. https://github.com/creationix/nvm

**Comment by Chris-Johnston** [comment]
You are right, I installed node using my OS' package manager (I am also not very familiar with node.js):
```
$ node --version
v10.6.0
$ which node
/usr/bin/node
```

It looks like nvm did the trick.

I followed these steps to get it working:
1. Uninstall package manager version of node.js
2. Install nvm node.js
3. Re-install github-web-launcher
4. `$ sudo mkdir /usr/lib/node_modules/github-web-launcher/downloads` (directory still didn't exist)
5. `$ sudo chown chris:chris /usr/lib/node_modules/github-web-launcher/downloads`
6. Run `web` and let it download the files it needs.
